diff --git a/node_modules/expo-modules-core/ios/ReactDelegates/EXReactCompatibleHelpers.m b/node_modules/expo-modules-core/ios/ReactDelegates/EXReactCompatibleHelpers.m
index 50c0019..82fadc3 100644
--- a/node_modules/expo-modules-core/ios/ReactDelegates/EXReactCompatibleHelpers.m
+++ b/node_modules/expo-modules-core/ios/ReactDelegates/EXReactCompatibleHelpers.m
@@ -4,22 +4,22 @@
 
 #import <React/RCTRootView.h>
 
-#if __has_include(<React/RCTAppSetupUtils.h>)
+#if __has_include(<React-RCTAppDelegate/RCTAppSetupUtils.h>)
+#import <React-RCTAppDelegate/RCTAppSetupUtils.h>
+#elif __has_include(<React_RCTAppDelegate/RCTAppSetupUtils.h>)
+// for importing the header from framework, the dash will be transformed to underscore
+#import <React_RCTAppDelegate/RCTAppSetupUtils.h>
+#else
+// react-native < 0.72
 #import <React/RCTAppSetupUtils.h>
 #endif
 
 UIView *EXAppSetupDefaultRootView(RCTBridge *bridge, NSString *moduleName, NSDictionary *initialProperties, BOOL fabricEnabled)
 {
-#if __has_include(<React/RCTAppSetupUtils.h>)
-
 #if REACT_NATIVE_MINOR_VERSION >= 71
   return RCTAppSetupDefaultRootView(bridge, moduleName, initialProperties, fabricEnabled);
 #else
   return RCTAppSetupDefaultRootView(bridge, moduleName, initialProperties);
 #endif // REACT_NATIVE_MINOR_VERSION >= 71
-
-#else
-  return [[RCTRootView alloc] initWithBridge:bridge moduleName:moduleName initialProperties:initialProperties];
-#endif // __has_include(<React/RCTAppSetupUtils.h>)
 }
 
